debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/ShrCCommand.java

```diff
@@ -112,28 +115,61 @@ public void run(String command, String[] args, Context context, PrintStream out)
 						String addr;
 						
 						addr = args[1].substring(0, args[1].indexOf(rangeDelim));
-						metaStart = U8Pointer.cast(CommandUtils.parsePointer(addr, J9BuildFlags.env_data64));
+						metaStart[0] = U8Pointer.cast(CommandUtils.parsePointer(addr, J9BuildFlags.env_data64));
 						
 						addr = args[1].substring(args[1].indexOf(rangeDelim) + rangeDelim.length());
-						metaEnd = U8Pointer.cast(CommandUtils.parsePointer(addr, J9BuildFlags.env_data64));
+						metaEnd[0] = U8Pointer.cast(CommandUtils.parsePointer(addr, J9BuildFlags.env_data64));
+						userSpecRange = true;
+					} else if (args[1].indexOf("layer=") != -1) {
+						int topLayer = dbgShrcCacheTopLayer(out, sharedClassConfig);
+						if (-1 == topLayer) {
+							CommandUtils.dbgPrint(out, "This cache does not have layers\n");
+							return;
+						}
+						Pattern p = Pattern.compile("(.*)layer=(\\d+)$");
+						Matcher m = p.matcher(args[1]);
+						if (m.matches()) {
+							String layerNumberString = m.group(2);
+							layer = Integer.parseInt(layerNumberString);
+							if (layer >= metaStartInCache.length) {
+								CommandUtils.dbgPrint(out, "User specified layer number %d is invalid, the maximum layer number is %d\n", layer, metaStartInCache.length - 1);
+								return;
+							}
+							metaStart[0] = U8Pointer.cast(metaStartInCache[layer]);
+							metaEnd[0] = U8Pointer.cast(metaEndInCache[layer]);
+							userSpecRange = true;
+						}
 					}
-				}
-				
-				if (metaStart != null && metaStart.longValue() < metaStartInCache.longValue()) {
-					CommandUtils.dbgPrint(out, "User specified start boundary is before metadata start in cache\n");
+					if (userSpecRange) {
+						if ((null != metaStart[0])
+							&& (null != metaEnd[0])
+							&& (metaEnd[0].longValue() < metaStart[0].longValue())
```

Comment:
This should be `metaEnd[0].lt(metaStart[0])` to avoid problems with addresses larger than `Long.MAX_VALUE`.

Reply:
Fixed